src/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -127,8 +134,12 @@ const SwitchButton = styled.button<SwitchButtonProps>`
   }
 
   ${props => {
-    if (props.disabled) {
+    if (props['aria-disabled']) {
```

Comment:
we briefly chatted about ToggleSwitch with @siddharthkp today and one question we had was when the toggle is disabled, should we have still have visual cues? Also should the cursor be not allowed on the "On/off" label too?

Quick recording from Sid:

https://github.com/primer/react/assets/1446503/370289fc-4529-4ba9-8210-78af9a042c25

_Video description: Toggle switch disabled story is open and the mouse is moving on the toggle switch displaying not allowed cursor and subtle hover styles appear on the control as well. Then mouse moves over to the Off label and the cursor is pointer_

Reply:
I agree with @broccolinisoup, since this region is usually interactive, we'll want to add a visual cue (`cursor: not-allowed`) to indicate that it is currently unavailable when the component is disabled or loading.

Reply:
Thanks for adding the `cursor: not-allowed` to the label as well @camertron! I think the hover state change is not needed when the switch is disabled. See the [PVC example for reference](https://view-components-storybook.eastus.cloudapp.azure.com/view-components/lookbook/inspect/primer/alpha/toggle_switch/disabled).

```
 &:hover:not([aria-disabled]),
  &:focus:focus-visible:not([aria-disabled]) {
    background-color: ${get('colors.switchTrack.hoverBg')};
  }
```

I think that should remove the hover background change when it is disabled. Let me know if you have any concerns!